drop-boxes/register-omero-metadata/register-omero.py

```diff
@@ -144,11 +157,75 @@ def findMetaDataFile(incomingPath):
 		for f in files:
 			stem, ext = os.path.splitext(f)
 			if ext.lower()=='.tsv':
-				with open(os.path.join(root, f), 'U') as fh: metadataFile = fh.readlines()
+				with open(os.path.join(root, f), 'U') as fh: metadataFileContent = fh.readlines()
 	return metadataFileContent
 
+def getPropertyNames(metadataFile):
+	"""Here we could add more complex behaviour later on.
+	"""
+	
+	property_names = metadataFile[0].split("\t")
+	for i in range(len(property_names)):
+		property_names[i] = property_names[i].strip().upper()
+
+	return property_names
+
+def validatePropertyNames(property_names):
+	"""Validate metadata property names.
+	TODO: call the imaging metadata parser (with json schema).
+	"""
+
+	# fast validation without parser object.
+	required_names = ["IMAGE_FILE_NAME", "IMAGING_MODALITY", "IMAGED_TISSUE", "INSTRUMENT_MANUFACTURER", "INSTRUMENT_USER", "IMAGING_DATE"]
+
+	for name in required_names:
+		if not name in property_names:
+			return False
+
+	return True
+
+def getPropertyMap(line, property_names):
+	"""Build the property map. Here we could add more complex behaviour later on.
+	"""
+
+	properties = {}
+	property_values = line.split("\t")
+
+	for i in range(1, len(property_names)): #exclude first col (filename)
+		##remove trailing newline, and replace space with underscore
+		name = property_names[i].rstrip('\n').replace(" ", "_")
+		value = property_values[i].rstrip('\n').replace(" ", "_")
+
+		properties[name] = value
+
+	return properties
+
+def filterOmeroPropertyMap(property_map):
+	"""Filters map before ingestion into omero server
+	"""
+
+	#the blacklist, e.g. what is going to openBIS or is automatically added to omero (e.g. file name)
+	filter_list = ["IMAGE_FILE_NAME", "INSTRUMENT_USER", "IMAGING_DATE"]
+
+	new_props = {}
+	for key in property_map.keys():
+		if not key in filter_list:
+			new_props[key] = property_map[key]
+
+	return new_props
+
+
+def printPropertyMap(property_map):
+	"""Function to display metadata properties.
+	"""
+
+	print("KEY : VALUE")
+	for key in property_map.keys():
+		print "--> " + str(key) + " : " + str(property_map[key])
+
```

Comment:
Please remove that for production logging please. This just causes log flooding.

Reply:
I would remove the call to the function, but leave the function itself, it might be useful for fixing issues in the future

Reply:
done

Reply:
pushed? i still see it somehow

Reply:
as mentioned, I removed the call to the function only